biggraphite/metadata_cache.py

```diff
@@ -400,8 +417,14 @@ def repair(self, start_key=None, end_key=None, shard=0, nshards=1):
                     continue
 
                 metric = self._accessor.get_metric(key)
-                expected_value = self.__get_value(metric) if metric else None
-                if value != expected_value:
```

Comment:
maybe factorize the code to split somewhere, and compare only what you want (uuid + json). This way to isolate the parsing part of the code